safesocket/safesocket_darwin.go

```diff
@@ -46,6 +47,15 @@ func localTCPPortAndTokenMacsys() (port int, token string, err error) {
 	if auth == "" {
 		return 0, "", errors.New("empty auth token in sameuserproof file")
 	}
+
+	// The above files exist forever after the first run. Check that the server
+	// is actually running (though it may still be in a Stopped state).
+	conn, err := net.Dial("tcp", "127.0.0.1:"+portStr)
```

Comment:
We could try even harder than this by making a HTTP request and looking for Tailscale headers:

```
$ curl -si http://127.0.0.1:61115/ | grep ^Tailscale
Tailscale-Cap: 68
Tailscale-Version: 1.48.1-t528f95da6-gc4e4acad7
```

or checking that it's a process called Tailscale doing the listening:

```
$ lsof -i:61115 -F | grep ^cTailscale
cTailscale
```

Reply:
If we're doing an L7 check, we could also confirm validity of the auth token :) But realistically it's pretty unlikely that something else will be listening on a randomly chosen port, so I think a simple TCP check is fine.